kmir/src/kmir/kdist/mir-semantics/symbolic/spl-token.md

```diff
@@ -504,6 +547,31 @@ expose the wrapped payload directly.
         => #setLocalValue(DEST, SPLRefCell(PLACE, VAL)) ~> #continueAt(TARGET)
        ...
       </k>
+
+  // Cross-frame write helper for SPL ref cells
+  rule <k> #writeThroughRef(Reference(0, place(local(I), PROJS), _, _), VAL)
+        => #forceSetPlaceValue(place(local(I), PROJS), VAL)
+       ...
+      </k>
+      <locals> LOCALS </locals>
+    requires 0 <=Int I andBool I <Int size(LOCALS)
+     andBool isTypedLocal(LOCALS[I])
+
+  rule <k> #writeThroughRef(Reference(OFFSET, place(local(I), PROJS), _, _), VAL)
+        => #traverseProjection(
+             toStack(OFFSET, local(I)),
+             #localFromFrame({STACK[OFFSET -Int 1]}:>StackFrame, local(I), OFFSET),
+             PROJS,
+             .Contexts
+           )
+           ~> #writeProjection(VAL)
+       ...
+      </k>
+      <stack> STACK </stack>
+    requires 0 <Int OFFSET
+     andBool OFFSET <=Int size(STACK)
+     andBool isStackFrame(STACK[OFFSET -Int 1])
+     andBool 0 <=Int I
```

Comment:
The `#writeThroughRef` only operates on `Reference` values. It seems the `WriteTo` helper sort from `data.md` could be used instead of the first `Value` argument.
Many symbol arguments have changed sort from  `Place` to `Value`, was this the reason?